### PR TITLE
Fixed error for undefined

### PR DESCRIPTION
**Pull request by iamibi**
https://trello.com/c/5WiwHhF1/15-fix-errors-in-js

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -35,22 +35,22 @@ function startVideo() {
...
 video.addEventListener('play', () => {
-  
+
   setInterval(async () => {
-    const detections = await faceapi.detectAllFaces(video, 
+    const detections = await faceapi.detectAllFaces(video,
       new faceapi.TinyFaceDetectorOptions())
       .withFaceExpressions()
-    if (detections[0].expressions.happy > 0.3)
+    if (detections && detections[0] && detections[0].expressions.happy > 0.3)
```
Comment (ashshdwvdi) [inline script.js:43]: can expressions and happy property be nil as well ?

File: script.js [hunk 2]
```diff
@@ -35,22 +35,22 @@ function startVideo() {
...
 
 function happyFacedetected() {
   let currentScore = parseInt(document.getElementById('score').innerText);
   if (currentScore == 0) {
-    snackNotif(firstLaugh[parseInt(Math.random() * firstLaugh.length)]);  
-  }  
+    snackNotif(firstLaugh[parseInt(Math.random() * firstLaugh.length)]);
+  }
   else
     snackNotif(Laugh[parseInt(Math.random() * Laugh.length)]);
```
Comment (ashshdwvdi) [inline script.js:55]: please save `parseInt(Math.random() * firstLaugh.lenght` in a var we should do this operation only once may be put it in a separate function maybe and use bot in the if...else blocks ?

**Comment by ashshdwvdi** [comment]
one questions why we need to multiply random() with first laugh length ? @harshmandalgi  @iamibi ??

File: script.js (hunk 1 above)
Comment (harshmandalgi) [reply script.js:43]: I am not sure but I think they'll carry 0 as a value at minimum.

**Comment by harshmandalgi** [comment]
> one questions why we need to multiply random() with first laugh length ? @harshmandalgi @iamibi ??

Math.random gives us a random num between 0 and 1, when we multiply it with a limit(say 5), we can get a random number between 0 and 5

**Comment by ashshdwvdi** [comment]
> > one questions why we need to multiply random() with first laugh length ? @harshmandalgi @iamibi ??
> 
> Math.random gives us a random num between 0 and 1, when we multiply it with a limit(say 5), we can get a random number between 0 and 5

if its random can we not just multiply it by 10, since multiplication operation is costlier we can remove its usage 
?

**Comment by harshmandalgi** [comment]
> > > one questions why we need to multiply random() with first laugh length ? @harshmandalgi @iamibi ??
> > 
> > 
> > Math.random gives us a random num between 0 and 1, when we multiply it with a limit(say 5), we can get a random number between 0 and 5
> 
> if its random can we not just multiply it by 10, since multiplication operation is costlier we can remove its usage
> ?

multiplying by 10 means we are hardcoding the length of any array(firstLaugh, noLaugh, etc) with 10,  we need to fetch an element index between 0 and that array length. Thats why we multiply random num(between 0 and 1) with array length

File: script.js (hunk 2 above)
Comment (harshmandalgi) [reply script.js:55]: firstLaugh.length will have to passed to that function then, I think its a simple operation and we don't repeat it often so this looks fine to me

**Review by harshmandalgi — APPROVED** [review]
looks fine, please address Ashish's comments and merge @iamibi 
Thanks

File: script.js (hunk 2 above)
Comment (mrgreen-creates) [reply script.js:55]: it should be fine but should keep the logic at one place

**Review by mrgreen-creates — APPROVED** [review]

**Pull request merged**